### ISSUE TITLE
Remove can-define usage in CanJS core

### ISSUE THREAD

**Issue by phillipskevin** [canjs/canjs#5207]
Although, `can-define` was removed from `core.js`, the core build still included `can-define` because it is a dependency of a few other `core` packages:

- [can-rest-model](https://github.com/canjs/can-rest-model/blob/90917099932d1b394081beebf7fd4bbc89138ab1/can-rest-model.js#L16-L21)
- [can-realtime-rest-model](https://github.com/canjs/can-realtime-rest-model/blob/8f05f16e2d0655f317cd84a9b45b7db32edd5411/can-realtime-rest-model.js#L20-L26)
- [can-route/routedata](https://github.com/canjs/can-route/blob/dadda0b603edbbd8177e99d8eaab0a08b9295d67/src/routedata.js#L4)

We should consider changing these to use `ObservableObject` and `ObservableArray`.

## Benefits

1. Documentation will be more consistent

Things like this (from the homepage) mention `Map`, where everything else is using `Object`/`Array`:

![image](https://user-images.githubusercontent.com/5851984/63048347-fa4bf100-be9b-11e9-8d9d-25923d0fabeb.png)

2. Including `can-define` in core makes the build larger

## Open questions

1. Would this break IE11 support for people using `can-component` and `can-define`?
  a. Should we run more of the test suite in IE11? (currently we only run the steal development tests)
2. Would this break users' apps in unexpected ways? (missing methods like `get`,`set`, etc)

**Comment by matthewp** [comment]
It's also in can-component. I'd probably change can-component first and run against IE11 and see what happens. I think that's probably the biggest risk.

**Comment by phillipskevin** [comment]
But `can-component` is no longer in core in 6.0. We want people to be able to continue using `can-component`/`can-define` if they need to support IE11.

**Comment by phillipskevin** [comment]
We are going to do this for 6.0. For the open questions:

1. We will provide `-legacy` packages that are needed for people to keep using `can-component` and `can-define` based apps in IE11
2. These are unlikely as people will only be using `ObservableObject` instead of `DefineMap` if they don't provide their own `Map` option and don't use the `-legacy` packages. In these cases, they can use one of these work-arounds. We can also add these methods to `ObservableObject` later if we decide to do that.

**Comment by chasenlehara** [comment]
Re: open question 1, has it already been discussed/decided that we should create `-legacy` packages vs. `/legacy` modules (e.g. `can-rest-model/legacy`)? I would expect the latter to be an easier transition, but I’m sure there are good reasons for making a separate package instead.

Assuming we make separate packages, is the best thing to do just to fork the current package and set everything up to have the legacy name? Any gotchas here?

**Comment by chasenlehara** [comment]
Right now `can-realtime-rest-model` and `can-rest-model` take `List` and `Map` as options, and also expose those on the connection.

I see a few different options:

## 1. Keep `List` and `Map`

Pros:

- Less code for users to change/migrate.

Cons:

- The `Map` name is held over from `DefineMap` and doesn’t make as much sense with `can-observable-object`
- `Map` might need to be assigned a different name for destructuring

Unknowns (for me):

- Would `can-connect` need to do more work to support both types using the same option name?

## 2. Use `Array` and `Object` instead

Pros:

- Consistent with `can-define-array` and `can-define-object`

Cons:

- `Array` and `Object` need to be assigned different names for destructuring
- `getList` is still a thing, so this adds a little naming inconsistency

Unknowns (for me):

- Would we add support for `ObservableObject.Array` being the list type? Is this already a thing anywhere in the code?

## 3. Use `ArrayType` and `ObjectType` instead

Basically the same as option 2, but:

Pros:

- No destructuring issues

Cons:

- Longer names

## 4. Keep `List` and rename `Map` to `Model`

Pros:

- Names don’t have to be reassigned during destructuring

Unknowns (for me): all the same unknowns as the above options

**Comment by phillipskevin** [comment]
> has it already been discussed/decided that we should create -legacy packages vs. /legacy

I don't think a final decision was made. Either would be fine as long as we can ensure that `can-define` is not in the build if you're not using it.
